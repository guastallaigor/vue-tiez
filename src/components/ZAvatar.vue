<template>
  <span class="z-avatar" :class="[size, {'dark': dark, 'stacked': stacked}]">
    <img :alt="alt" :title="alt" :src="src" />
  </span>
</template>

<script>
export default {
  name: 'z-avatar',
  props: {
    src: {
      type: String,
      required: true
    },
    alt: {
      type: String,
      default: ''
    },
    size: {
      type: String,
      default: '',
      validator: (v) => (['small', '', 'medium', 'large'].includes(v))
    },
    dark: {
      type: Boolean,
      default: false
    },
    stacked: {
      type: Boolean,
      default: false
    },
    show: {
      type: Number,
      default: 0
    }
  }
}
</script>

<style lang="scss">
@mixin size($width: 20px, $height: 20px) {
  width: $width;
  height: $height;
}

.z-avatar {
  &.small {
    @include size();
  }

  @include size(30px, 30px);

  &.medium {
    @include size(60px, 60px);
  }

  &.large {
    @include size(90px, 90px);
  }

  img {
    width: 100%;
    height: 100%;
  }

  display: inline-flex;
  align-items: center;
  line-height: 0;
  vertical-align: top;
  border-radius: 100%;
  overflow: hidden;
  border-width: 1px;
  border-style: solid;
  border-color: rgb(238, 238, 238);
  border-image: initial;
  background: rgb(255, 255, 255);
  -webkit-mask-image: -webkit-radial-gradient(center, circle cover, white, black);

  &.dark {
    border-color: rgb(51, 51, 51);
    background: rgb(0, 0, 0);
  }

  &.stacked {
    margin-left: -10px;
  }
}
</style>
