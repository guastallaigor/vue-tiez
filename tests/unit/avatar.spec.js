import { shallowMount } from '@vue/test-utils'
import ZAvatar from '../../src/components/ZAvatar'

describe('When I create the ZAvatar component', () => {
  const createComponent = (propsData = {}, slot) => {
    const props = { ...propsData }
    props.src = props.src || 'https://ik.imagekit.io/6xhf1gnexgdgk/igor2_HJhiHMa54.png'

    return shallowMount(ZAvatar, {
      propsData: props,
      slots: {
        default: slot || 'ZAvatar'
      }
    })
  }

  it('should be a Vue instance', () => {
    const wrapper = createComponent()
    expect(wrapper.find(ZAvatar).isVueInstance()).toBe(true)
  })

  it('should have an img with the src attribute', () => {
    const wrapper = createComponent()
    const avatar = wrapper.find('.z-avatar')
    expect(avatar.exists()).toBe(true)
    expect(avatar.classes().length).toBe(1)
    expect(avatar.classes()).toEqual(['z-avatar'])
    const img = wrapper.find('.z-avatar > img')
    expect(img.exists()).toBe(true)
    expect(img.attributes().src).toBe('https://ik.imagekit.io/6xhf1gnexgdgk/igor2_HJhiHMa54.png')
  })

  it('should have an alt when the alt prop is passed', () => {
    const wrapper = createComponent({ alt: 'Photo of Igor' })
    const avatar = wrapper.find('.z-avatar')
    expect(avatar.exists()).toBe(true)
    const img = wrapper.find('.z-avatar > img')
    expect(img.exists()).toBe(true)
    expect(img.attributes().alt).toBe('Photo of Igor')
  })

  it('should have a dark class when the dark prop is true', () => {
    const wrapper = createComponent({ dark: true })
    const avatar = wrapper.find('.z-avatar')
    expect(avatar.exists()).toBe(true)
    expect(avatar.classes().length).toBe(2)
    expect(avatar.classes()).toEqual(['z-avatar', 'dark'])
  })

  it('should have a stacked class when the stacked prop is true', () => {
    const wrapper = createComponent({ stacked: true })
    const avatar = wrapper.find('.z-avatar')
    expect(avatar.exists()).toBe(true)
    expect(avatar.classes().length).toBe(2)
    expect(avatar.classes()).toEqual(['z-avatar', 'stacked'])
  })

  it('should have 4 different sizes when the size prop is passed', () => {
    const wrapper = createComponent({ size: 'small' })
    const avatar = wrapper.find('.z-avatar')
    expect(avatar.exists()).toBe(true)
    expect(avatar.classes().length).toBe(2)
    expect(avatar.classes()).toEqual(['z-avatar', 'small'])
    wrapper.setProps({ size: 'medium' })
    expect(avatar.exists()).toBe(true)
    expect(avatar.classes().length).toBe(2)
    expect(avatar.classes()).toEqual(['z-avatar', 'medium'])
    wrapper.setProps({ size: 'large' })
    expect(avatar.exists()).toBe(true)
    expect(avatar.classes().length).toBe(2)
    expect(avatar.classes()).toEqual(['z-avatar', 'large'])
    wrapper.setProps({ size: '' })
    expect(avatar.exists()).toBe(true)
    expect(avatar.classes().length).toBe(1)
    expect(avatar.classes()).toEqual(['z-avatar'])
  })

  it('should match snapshot', () => {
    const wrapper = createComponent({
      src: 'https://ik.imagekit.io/6xhf1gnexgdgk/igor2_HJhiHMa54.png',
      alt: 'Photo of Igor',
      size: '',
      dark: false,
      stacked: false
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
