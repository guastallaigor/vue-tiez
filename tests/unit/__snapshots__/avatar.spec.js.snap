// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I create the ZAvatar component should match snapshot 1`] = `<span class="z-avatar"><img alt="Photo of Igor" title="Photo of Igor" src="https://ik.imagekit.io/6xhf1gnexgdgk/igor2_HJhiHMa54.png"></span>`;
